client/lib/mixins/close-on-esc/hoc.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import { filter, last } from 'lodash';
+import React, { PureComponent } from 'react';
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'esc-hoc' );
+let components = [];
+
+function startCloseOnEscForComponent( component ) {
+	components.push( component );
+
+	debug( 'start', components.length );
+
+	if ( components.length > 0 ) {
```

Comment:
If more than one component is rendered to the page, it seems we add an event handler for each, when in fact it seems we only really need a maximum one?